app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -79,13 +79,16 @@ def update
     param :description, String, :desc => N_("Description for the new published content view version")
     param :force_yum_metadata_regeneration, :bool, :desc => N_("Force yum metadata regeneration on the repositories " \
                                                            "in the content view version")
-    param :major, :number, :desc => N_("Override the major version number"), :required => false
-    param :minor, :number, :desc => N_("Override the minor version number"), :required => false
+    param :major, :number, :desc => N_("Override the major version number")
+    param :minor, :number, :desc => N_("Override the minor version number")
+    param :repos_units, Hash, :desc => N_("list of repositories and units to publish")
     def publish
+      repos_units = params[:repos_units] ? JSON.parse(params[:repos_units]) : nil
```

Comment:
why is the JSON.parse necessary?

Reply:
@jlsherrill without the parse, it will just keep the field as a string and not let me access the fields stored inside easily. I'm not sure if this is expected behavior or not.

Reply:
hrmmm, you sure its being passed to the api correctly?   what does the request look like?

Reply:
example: `http -j -a admin:changeme --verify=no POST https://localhost/katello/api/v2/content_views/4/publish major=5 minor=25 repos_units=@packages.json`

I confirmed that it's sending `Content-Type: application/json` in the header, I feel like this should work without calling `JSON.parse` but not sure what's happening.

Reply:
i think apipie there is sending repo_units as a string, such as:

{ "repo_units": "[\"this should be json, but is actually a string\"]" }